pkg/yurthub/proxy/local/faketoken.go

```diff
@@ -0,0 +1,125 @@
+			headerN, _ := strconv.Atoi(headerNStr)
+			n, err := buf.ReadFrom(req.Body)
+			if err != nil || (headerN != 0 && int(n) != headerN) {
+				klog.Warningf("read body of post request when cluster is unhealthy, expect %d bytes but get %d bytes with error, %v", headerN, n, err)
+			}
+
+			s := createSerializer(req, tokenRequestGVR, serializerManager)
+			if s == nil {
+				klog.Errorf("skip fake token inject for request %s when cluster is unhealthy, failed to create serializer.", util.ReqString(req))
+				writeRequestDirectly(w, req, buf.Bytes(), n)
+				return
+			}
+
+			tokenRequset, err := getTokenRequestWithFakeToken(buf.Bytes(), req, s)
+			if err != nil {
+				klog.Errorf("skip fake token inject for request %s when cluster is unhealthy, failed to get token request: %v", util.ReqString(req), err)
+				writeRequestDirectly(w, req, buf.Bytes(), n)
+				return
+			}
+
+			klog.Infof("write fake token for request %s when cluster is unhealthy", util.ReqString(req))
+			err = util.WriteObject(http.StatusCreated, tokenRequset, w, req)
+			if err != nil {
+				klog.Errorf("write fake token resp for token request when cluster is unhealthy with error, %v", err)
+			}
+			return
+		}
+
+		handler.ServeHTTP(w, req)
+	})
+}
+
+func createSerializer(req *http.Request, gvr schema.GroupVersionResource, sm *serializer.SerializerManager) *serializer.Serializer {
+	ctx := req.Context()
+	reqContentType, _ := util.ReqContentTypeFrom(ctx)
+	respContentType := reqContentType
+	return sm.CreateSerializer(respContentType, gvr.Group, gvr.Version, gvr.Resource)
+}
+
+func writeRequestDirectly(w http.ResponseWriter, req *http.Request, data []byte, n int64) {
+	copyHeader(w.Header(), req.Header)
+	w.WriteHeader(http.StatusCreated)
+	nw, err := w.Write(data)
+	if err != nil || nw != int(n) {
+		klog.Errorf("write resp for token request when cluster is unhealthy, expect %d bytes but write %d bytes with error, %v", n, nw, err)
+	}
+}
+
+func getTokenRequestWithFakeToken(data []byte, req *http.Request, s *serializer.Serializer) (*authv1.TokenRequest, error) {
+	obj, err := s.Decode(data)
+	if err != nil || obj == nil {
+		return nil, errors.Errorf("decode reuqest with error %v", err)
+	}
+	if tokenRequest, ok := obj.(*authv1.TokenRequest); ok {
+		token, err := getFakeToken()
+		if err != nil {
+			return nil, errors.Wrapf(err, "get fake token")
+		}
+		klog.V(5).Infof("the fake token of request %s is %s", util.ReqString(req), token)
+		tokenRequest.Status = authv1.TokenRequestStatus{
+			Token:               token,
+			ExpirationTimestamp: metav1.NewTime(time.Now().Add(1 * time.Minute)),
+		}
+		return tokenRequest, nil
+	}
+
+	return nil, errors.Errorf("trans request object to token request failed.")
+}
+
+func getFakeToken() (string, error) {
+	expireTime := time.Now().Add(1 * time.Minute)
+	claims := &jwt.StandardClaims{
+		ExpiresAt: expireTime.Unix(),
+		IssuedAt:  time.Now().Unix(),
+		Issuer:    "openyurt",
+		Subject:   "autonomy",
+	}
```

Comment:
sorry, the comment comes late, how about to construct claims based on the namespace and name?  and you can reference K8s code here: https://github.com/kubernetes/kubernetes/blob/master/pkg/serviceaccount/claims.go#L63-L69

because `WithSaTokenSubstitute` handler in yurthub(https://github.com/openyurtio/openyurt/blob/master/pkg/yurthub/proxy/util/util.go#L304-L335) need to use subject in bearer token.

Reply:
OK